packages/appsync-modelgen-plugin/src/visitors/appsync-visitor.ts

```diff
@@ -1028,7 +1028,9 @@ export class AppSyncModelVisitor<
             connectionInfo &&
             connectionInfo.kind !== CodeGenConnectionType.HAS_MANY &&
             connectionInfo.kind !== CodeGenConnectionType.HAS_ONE &&
-            connectionInfo.targetName !== 'id'
+            connectionInfo.targetName !== 'id' &&
+            !(this.config.target === 'introspection' && 
```

Comment:
What are the other targets that are valid here and why this fix only required for "introspection" target?

Reply:
The other targets are controlled by the feature flag value. For introspection, there are cases where the app has CPK value as false and the code block above is entered. That's why it is protected not to execute the line when target is introspection and belongsTo connnection field is primary key